### PR TITLE
server: support a simple region split function

### PR DESCRIPTION
**Pull request by hanfei1991**
Signed-off-by: Han Fei <hanfei19910905@gmail.com>

Implement Region Split by GRPC interface.

### CODE REVIEW

File: server/cluster.go [hunk 1]
```diff
@@ -189,6 +189,16 @@ func (c *clusterInstance) allocID() uint64 {
 	return c.server.allocID()
 }
 
+func (c *clusterInstance) allocIDs(length int) []uint64 {
+	ids := make([]uint64, 0, length)
+	i := 0
+	for i < length {
```
Comment (disksing) [inline server/cluster.go]: for i := 0; i < length; i++ {
}

File: server/cluster.go [hunk 2]
```diff
@@ -295,3 +305,26 @@ func (c *clusterInstance) deleteStoreFailPoint(storeID uint64, failPoint string)
...
+	c.Lock()
+	defer c.Unlock()
+	originRegion := c.regionByID[oldRegionID]
+	if originRegion == nil {
+		return nil, nil, errRegionNotFound
+	}
+	newRegion := newRegionInstance(newRegionID, splitKey, originRegion.EndKey)
+	newRegion.RegionEpoch = &metapb.RegionEpoch{
+		ConfVer: c.allocID(),
+		Version: c.allocID(),
```
Comment (disksing) [inline server/cluster.go]: Use tikv behavior: `ConfVer = origin.ConfVer, Version = origin.Version+1`

File: server/cluster.go (hunk 1 above)
Comment (hanfei1991) [reply server/cluster.go]: done

File: server/cluster.go (hunk 2 above)
Comment (hanfei1991) [reply server/cluster.go]: done

**Review by disksing — APPROVED** [review]

**Pull request merged**